Sprint 6 solution

alsygaev opened the pull request:
Провел рефакторинг в соответствии с требованиями ФЗ-6

alsygaev committed:
Провел рефакторинг кода в соответствии с требованиями ФЗ-6

alsygaev committed:
Провел рефакторинг кода в соответствии с требованиями ФЗ-6

alsygaev committed:
Remove unused import in Epic.java

alsygaev committed:
Fix formatting issues in HistoryManager.java

alsygaev committed:
Fix formatting issues in Managers.java

VadimZharkov commented on src/InMemoryTaskManager.java:
```diff
@@ -127,44 +127,75 @@ public void updateSubtask(Subtask subtask) {
+
+        historyManager.getHistory()
+                .stream()
+                .filter(task -> task instanceof Task)
```
С точки зрения ООП использование instanceof считается плохой практикой. Лучше когда знание об объекте инкапсулированы в него самого. Поэтому предлагаю у сделать перечислимое с типами тасков и в класс Task добавить getType(), в наследниках переопределить. Далее везде вместо instanceof использовть task.getType().

Можно по другому поступить. Пройтись сначала по мапе и удалить из истории таски содержащиеся в мапе, а уже потом  делать tasks.clear();

VadimZharkov reviewed (changes requested):
Хорошая работа, можно улучшить.

alsygaev committed:
Исправил замечания ревью:
 *Пройтись сначала по мапе и удалить из истории таски содержащиеся в мапе, а уже потом делать tasks.clear();

alsygaev committed:
Исправил замечания ревью:
 *Пройтись сначала по мапе и удалить из истории таски содержащиеся в мапе, а уже потом делать tasks.clear();
